src/domain/login/PasswordLoginViewModel.js

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {ViewModel} from "../ViewModel.js";
+import {LoginFailure} from "../../matrix/SessionContainer.js";
+
+export class PasswordLoginViewModel extends ViewModel {
+    constructor(options) {
+        super(options);
+        const {loginOptions, sessionContainer, homeserver, attemptLogin, showError} = options;
+        this._sessionContainer = sessionContainer;
+        this._loginOptions = loginOptions;
+        this._attemptLogin = attemptLogin;
+        this._showError = showError;
+        this._homeserver = homeserver;
+        this._isBusy = false;
+    }
+
+    get isBusy() { return this._isBusy; }
+
+    _toggleBusy(state) {
+        this._isBusy = state;
+        this.emitChange("isBusy");
```

Comment:
It might be better if the `LoginViewModel` indicates to all child view(models) that a login is being attempted. This would make it easy, for example, to disable the sso login button when a password login is being attempted.